lib/src/react16_suggestors/dependency_override_updater.dart

```diff
@@ -0,0 +1,187 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:codemod/codemod.dart';
+import 'package:meta/meta.dart';
+import 'package:source_span/source_span.dart';
+import 'package:yaml/yaml.dart';
+
+import '../constants.dart';
+
+/// Suggestor that adds overrides for over_react and react in the pubspec.
+class DependencyOverrideUpdater implements Suggestor {
+  /// Regex that matches the `dependency_overrides:` key in a pubspec.yaml.
+  static final RegExp dependencyOverrideKey = dependencyOverrideRegExp;
+
+  @override
+  Iterable<Patch> generatePatches(SourceFile sourceFile) sync* {
+    final contents = sourceFile.getText(0);
+    final dependencyOverrideSectionKey =
+        dependencyOverrideKey.firstMatch(contents);
+
+    // This needs to be kept track of because the patches are applied all at
+    // once, so even though we will add the dependency_override key the file
+    // won't be aware until all the patching is done.
+    bool shouldAddDependencyOverrideKey = dependencyOverrideSectionKey == null;
+
+    // Each override has its own object to keep track of what the dependency
+    // override is and what it needs to be overridden with.
+    // TODO update these versions to the dev branch after major release.
+    final reactDependencyOverride = DependencyOverrideItem(
```

Comment:
#nit 
not necessary but you could now that master is merged in:

`final reactDependencyOverride = DependencyCreator('react', gitOverride: 'https://github.com/cleandart/react-dart.git', ref: '5.0.0-wip');`